pkgs/applications/editors/vscode/vscodium.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, lib, callPackage, fetchurl, fetchpatch, ... } @ args:
+
+callPackage ./generic.nix (args // rec {
+
+  # How can I get plat and archive_fmt from generic.nix?
+  #inherit plat archive_fmt;
+
+  plat = {
+    "i686-linux" = "linux-ia32";
+    "x86_64-linux" = "linux-x64";
+    "x86_64-darwin" = "darwin";
+  }.${system};
+
+  archive_fmt = if system == "x86_64-darwin" then "zip" else "tar.gz";
+
+  inherit (stdenv.hostPlatform) system;
+
+  version = "1.33.1";
+
+  executableName = "vscodium";
+  longName = "VSCodium";
+  shortName = "Codium";
+
+  name = "vscodium-${version}";
+
+  sha256 = {
+    "i686-linux" = "c45be9ff32a5acef91befc85d545402daac5cef627fe8f17fea40ae387b35b22";
+    "x86_64-linux" = "7a87f19bc48cbe017321c72da244d879a7da364b92b3a89b742d437ffcd585d3";
+    "x86_64-darwin" = "e393be5dd82ea1900e3a0597b6cb2e4c924b6acda4297d9d3eff5abcc857f32c";
+  }.${system};
+
+  src = fetchurl {
+    name = "VSCodium_${version}_${plat}.${archive_fmt}";
```

Comment:
Not sure if `archive_fmt` is useful here, you seem to download .tar.gz in all cases.

Reply:
Not for darwin, hence:

```nix
archive_fmt = if system == "x86_64-darwin" then "zip" else "tar.gz";
```

Reply:
I look at the line just below
```
url = "https://github.com/VSCodium/vscodium/releases/download/${version}/VSCodium-${plat}-${version}.tar.gz";
```
for darwin evaluates to
```
url = "https://github.com/VSCodium/vscodium/releases/download/1.33.1/VSCodium-darwin-1.33.1.tar.gz";
```
and that is a broken link, so you need to use `archive_fmt` in your definition of `url`, and you can drop `name` attribute as it can be derived from url.

Reply:
oops, thanks :smile